libs/pika/async_mpi/include/pika/async_mpi/mpi_polling.hpp

```diff
@@ -224,6 +224,7 @@ namespace pika::mpi::experimental {
           : pool_name_(pool_name)
         {
             mpi::experimental::init(false, init_errorhandler);
+            mpi::experimental::register_polling();
```

Comment:
What happened here? Were we not registering polling before in the RAII helper? Should it be unregistered again in the destructor?

Reply:
AFAICR - the algorithm_transform_mpi test had to be changed to allow for pool creation etc and the polling RAII helper called init - but due to changes in the way we setup stuff, the polling was skipped, so I added it here - but when the RAII helper destructs, it calls `mpi::experimental::finalize(pool_name_);` which does unregister polling for us

Reply:
I see, thanks. Maybe we should separate those so that it's somewhat symmetric in the constructor and destructor (i.e. that one does explicit init+register and the other does explicit finalize+unregister). We don't need to change it here, but what do you think of this change?

Reply:
Also, it looks like the registering does not take into account the case that there's no `pool_name` set, while unregistering does. That should probably also be updated to be consistent? Again, not for this PR.